atc/db/check_factory.go

```diff
@@ -104,6 +104,10 @@ func (c *checkFactory) TryCreateCheck(ctx context.Context, checkable Checkable,
 		interval = *checkable.CheckEvery()
 	}
 
+	if len(resourceTypes) > 0 && !checkable.HasWebhook() && checkable.CheckEvery() == nil {
+		interval.Interval = atc.DefaultResourceTypeInterval
```

Comment:
From my testing it seemed to be okay, I was able to see in the logs that times were being set appropriately, maybe it might help to add a check to make sure the checkable isn't a resource as well just for extra robustness? It's been a few weeks so i'll need to have a look again to make sure that is definitely possible.

Reply:
I want to understand this check condition. Why `len(resourceTypes)` matters? `checkable.HasWebhook()` could also apply to a resource, so does `checkable.CheckEvery()`.

Maybe you can do:

```golang
     if _, ok := checkable.(resourceType); ok {
         interval.Interval = atc.DefaultResourceTypeInterval
     }
```

Reply:
I actually tried this out and I don't believe it worked as the interval never seemed to be updated, at least that I could see from the web logs(I added some logging here and there). Will make this back to [WIP] and try find a better solution.